### ISSUE TITLE
folders' unread counter counts unread topics instead of unread chats

### ISSUE THREAD

**Issue by RememberTheAir** [telegramdesktop/tdesktop#25867]
### Steps to reproduce

1. create a folder
2. manually include some chats and at least one topic chat
3. wait for some messages to be sent in one of the non-topic chats and in 2+ topics 

It's probably important to mention that I have the "count unread messages" option disabled, so the app counts chats instead of messages 

### Expected behaviour

The folder's unread counter counts the number of unique chats 

### Actual behaviour

The unread counter's count is the result of the sum of unread non-topic chats and the number of *single* unread topics. For example, let's say we have three unread chats in a folder: A, B, C. A and B are normal chats, while C is a topic chat, and there are three unread topics in this chat. The unread counter will show that there are 4 unread chats in this tab: 1 + 1 + 3

This might be the intended behavior on desktop, but it's inconsistent with mobile apps, where single chats are counted

### Operating system

Windows 10

### Version of Telegram Desktop

4.6.2

### Installation source

Static binary from official website

**Comment by SpL1Ne** [comment]
Also, if you don't check chat with topics long enough, it won't count as unread chat at all. I have 4 chats in folder, 1 chat with topics. And folder shows 3 unreaded chats instead of 4 (or instead 3+number of unreaded topics).

**Comment by RememberTheAir** [comment]
bump, this is still a distracting inconsistency with the mobile apps

**Comment by dajotim937** [comment]
Related issue with notification:
Screenshot 1 and 2: Chat with unread topics doesn't count in folder unread counter.
Screenshot 3: Open any topic, now we have unread counter for this topic
Screenshot 4: Now chat is counted in unread counter.

<details>
<summary>Screenshots</summary>

![2023-06-24_14-15-23](https://github.com/telegramdesktop/tdesktop/assets/96221151/9069fc82-e3c7-4635-aa8f-d873d35b7075)
![2023-06-24_14-15-31](https://github.com/telegramdesktop/tdesktop/assets/96221151/40d77724-419e-4040-bcfa-d9c7e3154b0d)
![2023-06-24_14-15-41](https://github.com/telegramdesktop/tdesktop/assets/96221151/1196c3c7-7c1c-4a57-93ec-59ef5d765644)
![2023-06-24_14-15-50](https://github.com/telegramdesktop/tdesktop/assets/96221151/fc2cac18-b5ba-438c-a9f4-88ff5ad07556)


</details>

**Comment by RememberTheAir** [comment]
this still works differently than the Android/iOS app. Is this intended?

**Comment by RememberTheAir** [comment]
any comment on this? it looks like it is not considered a bug, just wanted to make sure that the fact that from android my folder unread counter says 1 and on desktop it says 12 is intended or not

**Comment by RememberTheAir** [comment]
unstale, the mobile apps still behave differently

**Comment by kyak** [comment]
I just want to add that this issue is indeed very inconvenient.

**Comment by RememberTheAir** [comment]
yes, this needs to be fixed either here or on the mobile apps. @john-preston is this intended and going to stay this way?

**Comment by RememberTheAir** [comment]
any news? :(

**Comment by ciltocruz** [comment]
no news?

**Comment by 23rd** [comment]
This change can have a number of nuances, so it would be helpful if you tested a bit on how it works.

https://github.com/telegramdesktop/tdesktop/actions/runs/12012513096

**Comment by dbsergey** [comment]
Fixed for me

**Comment by dsent** [comment]
Commit f888008 seems to break counting of **muted** groups with topics (or groups with some topics muted, and some unmuted) in a lot of ways at the same time. The numbers seem to be subtracted from the unread count, not added. Additionally, triggers displaying of unread counters on folders as blue (active) even if all groups are muted. Makes things even more interesting if you disable counting of muted chats in folder badges.
With 5.8.3:
![image](https://github.com/user-attachments/assets/ccf1530f-211c-4a1f-adce-b97895137900)
With the commit in question:
![image](https://github.com/user-attachments/assets/34cdf16f-4d1a-4b3e-948e-258556239837)
With the muted counting disabled:
![image](https://github.com/user-attachments/assets/f6ebff54-cdcf-4d08-b3c5-3fb2a4866de2)

**Comment by dsent** [comment]
Having some topics muted, and some unmuted is a very real and important use case. We have three possible states of a group with topics, from the notification standpoint:
1. There are new messages in the unmuted topics of the group. The status for the whole group = +1 count, unmuted (blue).
2. No new messages in the unmuted topics, new messages in the muted topics. Group status = +1 count, muted (gray).
3. No new messages in any topics. Group status = 0

**Comment by dsent** [comment]
If counting messages is enabled, it gets messier. But still, you have to distinguish between having new messages in unmuted and muted topics, and give blue notifications only if there are new messages in the unmuted topics.

Sorry if I'm stating the obvious 🙏

**Comment by 23rd** [comment]
@dsent Thanks for your detailed feedback. Can you take a look at a new test build?
https://github.com/telegramdesktop/tdesktop/actions/runs/12044613567

**Comment by dsent** [comment]
@23rd this one seems to work correctly!
“Count unread messages” seems to affect only the window/taskbar title, not the folder badges, but I guess this was how it always worked.

**Comment by ciltocruz** [comment]
Sorry to butt in on the conversation.
I don't know if this issue is related to my question, if not, please let me know and I'll open a new one.

In the desktop app, the unread message counter in a folder is correct if the chats in that folder are private conversations or simple groups, but it is not correct if what is in that folder is a group with topics.

Let's suppose 1 folder with 1 group that has 10 topics. In all of them there are unread messages. It doesn't matter how many of those messages add up, the unread message counter will say that there are 10 unread messages (10 topics). To see the number of messages in each topic you have to go in and you never know, unless you manually count how many messages all the topics add up to.

I haven't seen an option in the app to count messages instead of topics.

My question is NOT related to the badge on the app which does mark the sum of all unread messages (including archived chats. That value is fine. I'm asking about the badge on the folders located in the sidebar)

**Comment by john-preston** [comment]
@ciltocruz Folders always show the number of chats, ignoring the "count unread messages" setting, that was done together with mobile apps, the same way counters work there. For now it works like that, intended.

**Comment by ciltocruz** [comment]
@john-preston you're right! Sorry for the question, so.

**Comment by dsent** [comment]
@Aokromes sorry, but the **original** issue is not about messages vs topics described above, and if you have to close it, it should probably be closed as fixed, not as "not planned". Because, well, it's fixed by @23rd's patches.
